TextToSpeech/TextToSpeechJsonRpc.cpp

```diff
@@ -166,17 +169,56 @@ namespace Plugin {
     {
         if(_tts) {
         Exchange::ITextToSpeech::Configuration config;
-        config.ttsEndPoint = GET_STR(parameters, "ttsendpoint", "");
-        config.ttsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        std::string proxyTtsEndPoint = GET_STR(parameters, "ttsendpoint", "");
+        std::regex urlPattern(R"((https?|ftp):\/\/\S+\/\S*)",std::regex::icase);
+        if(regex_match(proxyTtsEndPoint,urlPattern)) {
+            config.ttsEndPoint = proxyTtsEndPoint;
+        }
+        else {
+            config.ttsEndPoint = "";
+        }
+        std::string proxyTtsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        if(regex_match(proxyTtsEndPointSecured,urlPattern)) {
+            config.ttsEndPointSecured = proxyTtsEndPointSecured;
+        }
+        else {
+            config.ttsEndPointSecured = "";
+        }
         config.language = GET_STR(parameters, "language", "");
-        config.voice = GET_STR(parameters, "voice", "");
-        config.speechRate = GET_STR(parameters, "speechrate", "");
-        config.volume = (uint8_t) std::stod(GET_STR(parameters, "volume", "0.0"));
+        std::set<std::string> languageSet = {"en-US","es-MX","fr-CA","en-GB","de-DE","it-IT"};
+        config.language = languageSet.find(config.language) != languageSet.end() ? config.language : "";
+        std::string proxyVoice = GET_STR(parameters, "voice", "");
+        if(proxyVoice.c_str()[0] == ' ' || Utils::isValidUnsignedInt(const_cast<char*>(proxyVoice.c_str()))) {
```

Comment:
I think this validation is redundant (i.e checking if the incoming value is integral type when a string is expected. Instead, we need to have a list of supported voice (either populated from the config json or an entire list of possible voices) and match the incoming value against it